h/accounts/test/views_test.py

```diff
@@ -292,6 +293,264 @@ def test_logout_response_has_forget_headers(authn_policy):
+                                                   'dummy_db_session',
+                                                   'form_validator',
+                                                   'mailer',
+                                                   'routes_mapper',
+                                                   'user_model')
+
+
+@forgot_password_fixtures
+def test_forgot_password_returns_error_when_validation_fails(authn_policy,
+                                                             form_validator):
+    request = DummyRequest(method='POST')
+    authn_policy.authenticated_userid.return_value = None
+    form_validator.return_value = ({"errors": "KABOOM!"}, None)
+
+    result = ForgotPasswordController(request).forgot_password()
+
+    assert result == {"errors": "KABOOM!"}
+
+
+@forgot_password_fixtures
+def test_forgot_password_fetches_user_by_form_email(authn_policy,
+                                                    form_validator,
+                                                    user_model):
+    request = DummyRequest(method='POST')
+    authn_policy.authenticated_userid.return_value = None
+    form_validator.return_value = (None, {"email": "giraffe@thezoo.org"})
+
+    ForgotPasswordController(request).forgot_password()
+
+    user_model.get_by_email.assert_called_with(request, "giraffe@thezoo.org")
+
+
+@forgot_password_fixtures
+def test_forgot_password_creates_activation_for_user(activation_model,
+                                                     authn_policy,
+                                                     dummy_db_session,
+                                                     form_validator,
+                                                     user_model):
+    request = DummyRequest(method='POST')
+    authn_policy.authenticated_userid.return_value = None
+    form_validator.return_value = (None, {"email": "giraffe@thezoo.org"})
+
+    ForgotPasswordController(request).forgot_password()
+
+
+    user = user_model.get_by_email.return_value
+    activation = activation_model.return_value
+
+    activation_model.assert_called_with()
+    assert activation in dummy_db_session.added
+    assert user.activation == activation
+
+
+@patch('h.accounts.views.reset_password_link')
+@forgot_password_fixtures
+def test_forgot_password_generates_reset_link_from_activation(reset_link,
+                                                              activation_model,
+                                                              authn_policy,
+                                                              form_validator):
+    request = DummyRequest(method='POST')
+    authn_policy.authenticated_userid.return_value = None
+    form_validator.return_value = (None, {"email": "giraffe@thezoo.org"})
+    activation_model.return_value.code = "abcde12345"
+
+    ForgotPasswordController(request).forgot_password()
+
+    reset_link.assert_called_with(request, "abcde12345")
```

Comment:
Again, I'd say this test seems unnecessary given the test below it. The only reason that forgot_password() calls reset_password_link() is to pass the link on to reset_password_email(), and the test below tests that. It seems there's no way this test could be failing if the test below were not also failing.